services/fileuploader/fileuploader.go

```diff
@@ -54,14 +56,17 @@ func NewDefaultProvider() Provider {
 type provider struct {
 	onceInit        sync.Once
 	init            chan struct{}
+	mu              sync.RWMutex
 	storageSettings map[string]StorageSettings
 }
 
 func (p *provider) GetFileManager(workspaceID string) (filemanager.FileManager, error) {
 	<-p.init
+	p.mu.RLock()
+	defer p.mu.RUnlock()
 	settings, ok := p.storageSettings[workspaceID]
 	if !ok {
-		return nil, fmt.Errorf("no storage settings found for workspace: %s", workspaceID)
+		return nil, fmt.Errorf(noStorageForWorkspaceErrorString, workspaceID)
```

Comment:
we can introduce an unexported method to get storageSettings, so we don't have to:

```go
	<-p.init
	p.mu.RLock()
	defer p.mu.RUnlock()
```

In every method that requires them